src/plugins/sync-plugin.js

```diff
@@ -572,13 +588,25 @@ const normalizePNodeContent = pnode => {
   return res
 }
 
+const countYTextMarks = (yattrs) => {
+  let count = 0
+  object.forEach(yattrs, (val) => {
+    if (Array.isArray(val)) {
+      count += val.length
+    } else {
+      count++
+    }
+  })
+  return count
+}
+
 /**
  * @param {Y.XmlText} ytext
  * @param {Array<any>} ptexts
  */
 const equalYTextPText = (ytext, ptexts) => {
   const delta = ytext.toDelta()
-  return delta.length === ptexts.length && delta.every((d, i) => d.insert === /** @type {any} */ (ptexts[i]).text && object.keys(d.attributes || {}).length === ptexts[i].marks.length && ptexts[i].marks.every(mark => equalAttrs(d.attributes[mark.type.name] || {}, mark.attrs)))
+  return delta.length === ptexts.length && delta.every((d, i) => d.insert === /** @type {any} */ (ptexts[i]).text && countYTextMarks(d.attributes || {}) === ptexts[i].marks.length && ptexts[i].marks.every(mark => containsEqualMark(d.attributes[mark.type.name] || {}, mark.attrs)))
```

Comment:
Thanks for the PR for handling multiple marks!

I was pulling this PR into my project and ran into an issue with `containsEqualMark`. It looks like you're calling `containsEqualMark` with `yattrs, pattrs` instead of `pattrs, yattrs`.